{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "arn:aws:iam::065122976270:role/service-role/AmazonSageMaker-ExecutionRole-20200403T093426\n",
      "CPU times: user 851 ms, sys: 192 ms, total: 1.04 s\n",
      "Wall time: 12.8 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "import sagemaker\n",
    "from sagemaker import get_execution_role\n",
    "\n",
    "role = get_execution_role()\n",
    "print(role)\n",
    "sess = sagemaker.Session()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "bucket = 'sagemaker-object-detection-test-200408' # custom bucket name.\n",
    "# bucket = sess.default_bucket()\n",
    "prefix = 'ObjectDetection-v0'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "825641698319.dkr.ecr.us-east-2.amazonaws.com/object-detection:latest\n"
     ]
    }
   ],
   "source": [
    "from sagemaker.amazon.amazon_estimator import get_image_uri\n",
    "\n",
    "training_docker_image = get_image_uri(sess.boto_region_name, 'object-detection', repo_version=\"latest\")\n",
    "print (training_docker_image)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'s3://sagemaker-object-detection-test-200408/ObjectDetection-v0/output'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s3_output_location = 's3://{}/{}/output'.format(bucket, prefix)\n",
    "s3_output_location"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sagemaker configuration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "train_channel = prefix + '/train'\n",
    "validation_channel = prefix + '/validation'\n",
    "train_annotation_channel = prefix + '/train_annotation'\n",
    "validation_annotation_channel = prefix + '/validation_annotation'\n",
    "\n",
    "s3_train_data = 's3://{}/{}'.format(bucket, train_channel)\n",
    "s3_validation_data = 's3://{}/{}'.format(bucket, validation_channel)\n",
    "s3_train_annotation = 's3://{}/{}'.format(bucket, train_annotation_channel)\n",
    "s3_validation_annotation = 's3://{}/{}'.format(bucket, validation_annotation_channel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "od_model = sagemaker.estimator.Estimator(training_docker_image,\n",
    "                                         role, \n",
    "                                         train_instance_count=1, \n",
    "                                         train_instance_type='ml.p2.xlarge',\n",
    "                                         train_volume_size = 50,\n",
    "                                         train_max_run = 360000,\n",
    "                                         input_mode = 'File',\n",
    "                                         output_path=s3_output_location,\n",
    "                                         sagemaker_session=sess)\n",
    "# 'ml.p3.2xlarge',"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "od_model.set_hyperparameters(base_network='resnet-50',\n",
    "                             use_pretrained_model=1,\n",
    "                             num_classes=80,\n",
    "                             mini_batch_size=16,\n",
    "                             epochs=30,\n",
    "                             learning_rate=0.001,\n",
    "                             lr_scheduler_step='10',\n",
    "                             lr_scheduler_factor=0.1,\n",
    "                             optimizer='sgd',\n",
    "                             momentum=0.9,\n",
    "                             weight_decay=0.0005,\n",
    "                             overlap_threshold=0.5,\n",
    "                             nms_threshold=0.45,\n",
    "                             image_shape=512,\n",
    "                             label_width=600,\n",
    "                             num_training_samples=174)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "train_data = sagemaker.session.s3_input(s3_train_data, distribution='FullyReplicated', \n",
    "                        content_type='image/jpeg', s3_data_type='S3Prefix')\n",
    "validation_data = sagemaker.session.s3_input(s3_validation_data, distribution='FullyReplicated', \n",
    "                             content_type='image/jpeg', s3_data_type='S3Prefix')\n",
    "train_annotation = sagemaker.session.s3_input(s3_train_annotation, distribution='FullyReplicated', \n",
    "                             content_type='image/jpeg', s3_data_type='S3Prefix')\n",
    "validation_annotation = sagemaker.session.s3_input(s3_validation_annotation, distribution='FullyReplicated', \n",
    "                             content_type='image/jpeg', s3_data_type='S3Prefix')\n",
    "\n",
    "data_channels = {'train': train_data, 'validation': validation_data, \n",
    "                 'train_annotation': train_annotation, 'validation_annotation':validation_annotation}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-04-15 05:45:26 Starting - Starting the training job...\n",
      "2020-04-15 05:45:27 Starting - Launching requested ML instances...\n",
      "2020-04-15 05:46:22 Starting - Preparing the instances for training.........\n",
      "2020-04-15 05:47:45 Downloading - Downloading input data\n",
      "2020-04-15 05:47:45 Training - Downloading the training image........\u001b[34mDocker entrypoint called with argument(s): train\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:00 INFO 139779977344832] Reading default configuration from /opt/amazon/lib/python2.7/site-packages/algorithm/default-input.json: {u'label_width': u'350', u'early_stopping_min_epochs': u'10', u'epochs': u'30', u'overlap_threshold': u'0.5', u'lr_scheduler_factor': u'0.1', u'_num_kv_servers': u'auto', u'weight_decay': u'0.0005', u'mini_batch_size': u'32', u'use_pretrained_model': u'0', u'freeze_layer_pattern': u'', u'lr_scheduler_step': u'', u'early_stopping': u'False', u'early_stopping_patience': u'5', u'momentum': u'0.9', u'num_training_samples': u'', u'optimizer': u'sgd', u'_tuning_objective_metric': u'', u'early_stopping_tolerance': u'0.0', u'learning_rate': u'0.001', u'kv_store': u'device', u'nms_threshold': u'0.45', u'num_classes': u'', u'base_network': u'vgg-16', u'nms_topk': u'400', u'_kvstore': u'device', u'image_shape': u'300'}\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:00 INFO 139779977344832] Merging with provided configuration from /opt/ml/input/config/hyperparameters.json: {u'learning_rate': u'0.001', u'epochs': u'30', u'nms_threshold': u'0.45', u'optimizer': u'sgd', u'base_network': u'resnet-50', u'image_shape': u'512', u'label_width': u'600', u'lr_scheduler_step': u'10', u'momentum': u'0.9', u'overlap_threshold': u'0.5', u'num_training_samples': u'174', u'mini_batch_size': u'16', u'weight_decay': u'0.0005', u'use_pretrained_model': u'1', u'num_classes': u'80', u'lr_scheduler_factor': u'0.1'}\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:00 INFO 139779977344832] Final configuration: {u'label_width': u'600', u'early_stopping_min_epochs': u'10', u'epochs': u'30', u'overlap_threshold': u'0.5', u'lr_scheduler_factor': u'0.1', u'_num_kv_servers': u'auto', u'weight_decay': u'0.0005', u'mini_batch_size': u'16', u'use_pretrained_model': u'1', u'freeze_layer_pattern': u'', u'lr_scheduler_step': u'10', u'early_stopping': u'False', u'early_stopping_patience': u'5', u'momentum': u'0.9', u'num_training_samples': u'174', u'optimizer': u'sgd', u'_tuning_objective_metric': u'', u'early_stopping_tolerance': u'0.0', u'learning_rate': u'0.001', u'kv_store': u'device', u'nms_threshold': u'0.45', u'num_classes': u'80', u'base_network': u'resnet-50', u'nms_topk': u'400', u'_kvstore': u'device', u'image_shape': u'512'}\u001b[0m\n",
      "\u001b[34mProcess 1 is a worker.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:00 INFO 139779977344832] Using default worker.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:00 INFO 139779977344832] Loaded iterator creator application/x-image for content type ('application/x-image', '1.0')\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:00 INFO 139779977344832] Loaded iterator creator application/x-recordio for content type ('application/x-recordio', '1.0')\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:00 INFO 139779977344832] Loaded iterator creator image/png for content type ('image/png', '1.0')\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:00 INFO 139779977344832] Loaded iterator creator image/jpeg for content type ('image/jpeg', '1.0')\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:00 INFO 139779977344832] Checkpoint loading and saving are disabled.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:01 INFO 139779977344832] nvidia-smi took: 0.0753891468048 secs to identify 1 gpus\u001b[0m\n",
      "\u001b[34mCreating .rec file from /opt/ml/input/data/train/train.lst in /opt/ml/input/data/train\u001b[0m\n",
      "\u001b[34mtime: 0.0397000312805  count: 0\u001b[0m\n",
      "\u001b[34mCreating .rec file from /opt/ml/input/data/validation/val.lst in /opt/ml/input/data/validation\u001b[0m\n",
      "\u001b[34mtime: 0.0152831077576  count: 0\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:03 INFO 139779977344832] Number of GPUs being used: 1\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:03 WARNING 139779977344832] Training images are resized to image shape (3, 512, 512)\u001b[0m\n",
      "\u001b[34m[05:49:03] /opt/brazil-pkg-cache/packages/MXNetECL/MXNetECL-v1.4.1.1457.0/AL2012/generic-flavor/src/src/io/iter_image_det_recordio.cc:283: ImageDetRecordIOParser: /opt/ml/input/data/train/train.rec, use 3 threads for decoding..\u001b[0m\n",
      "\u001b[34m[05:49:03] /opt/brazil-pkg-cache/packages/MXNetECL/MXNetECL-v1.4.1.1457.0/AL2012/generic-flavor/src/src/io/iter_image_det_recordio.cc:340: ImageDetRecordIOParser: /opt/ml/input/data/train/train.rec, label padding width: 600\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:05 WARNING 139779977344832] Validation images are resized to image shape (3, 512, 512)\u001b[0m\n",
      "\u001b[34m[05:49:05] /opt/brazil-pkg-cache/packages/MXNetECL/MXNetECL-v1.4.1.1457.0/AL2012/generic-flavor/src/src/io/iter_image_det_recordio.cc:283: ImageDetRecordIOParser: /opt/ml/input/data/validation/val.rec, use 3 threads for decoding..\u001b[0m\n",
      "\u001b[34m[05:49:05] /opt/brazil-pkg-cache/packages/MXNetECL/MXNetECL-v1.4.1.1457.0/AL2012/generic-flavor/src/src/io/iter_image_det_recordio.cc:340: ImageDetRecordIOParser: /opt/ml/input/data/validation/val.rec, label padding width: 600\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:07 INFO 139779977344832] Number of GPUs being used: 1\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:07 INFO 139779977344832] Using [gpu(0)] as training context.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:07 INFO 139779977344832] Number of GPUs being used: 1\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:07 INFO 139779977344832] Create Store: device\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:07 INFO 139779977344832] Using (gpu(0)) as training context.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:07 INFO 139779977344832] Start training from pretrained model 1.\u001b[0m\n",
      "\u001b[34m[05:49:07] /opt/brazil-pkg-cache/packages/MXNetECL/MXNetECL-v1.4.1.1457.0/AL2012/generic-flavor/src/src/nnvm/legacy_json_util.cc:209: Loading symbol saved by previous version v0.8.0. Attempting to upgrade...\u001b[0m\n",
      "\u001b[34m[05:49:07] /opt/brazil-pkg-cache/packages/MXNetECL/MXNetECL-v1.4.1.1457.0/AL2012/generic-flavor/src/src/nnvm/legacy_json_util.cc:217: Symbol successfully upgraded!\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:07 INFO 139779977344832] Loaded pretrained model parameters.\u001b[0m\n",
      "\n",
      "2020-04-15 05:48:58 Training - Training image download completed. Training in progress.\u001b[34m[04/15/2020 05:49:13 INFO 139779977344832] Creating a new state instance.\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}}, \"EndTime\": 1586929753.591098, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"init_train_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\"}, \"StartTime\": 1586929753.59102}\n",
      "\u001b[0m\n",
      "\u001b[34m[05:49:13] /opt/brazil-pkg-cache/packages/MXNetECL/MXNetECL-v1.4.1.1457.0/AL2012/generic-flavor/src/src/operator/nn/./cudnn/./cudnn_algoreg-inl.h:97: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:56 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:56 INFO 139779977344832] #quality_metric: host=algo-1, epoch=0, batch=11 train cross_entropy <loss>=(4.6513203459692365)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:56 INFO 139779977344832] #quality_metric: host=algo-1, epoch=0, batch=11 train smooth_l1 <loss>=(0.7260459469986924)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:56 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:49:57 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:06 INFO 139779977344832] #quality_metric: host=algo-1, epoch=0, validation mAP <score>=(1.149756848897887e-06)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:06 INFO 139779977344832] Updating the best model with validation-mAP=1.149756848897887e-06\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:06 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:06 INFO 139779977344832] #progress_metric: host=algo-1, completed 3 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 1, \"sum\": 1.0, \"min\": 1}}, \"EndTime\": 1586929806.380902, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 0}, \"StartTime\": 1586929753.591349}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:25 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:25 INFO 139779977344832] #quality_metric: host=algo-1, epoch=1, batch=11 train cross_entropy <loss>=(4.3981388845261495)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:25 INFO 139779977344832] #quality_metric: host=algo-1, epoch=1, batch=11 train smooth_l1 <loss>=(0.7339986327347482)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:25 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:25 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:34 INFO 139779977344832] #quality_metric: host=algo-1, epoch=1, validation mAP <score>=(3.232645478487873e-05)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:34 INFO 139779977344832] Updating the best model with validation-mAP=3.232645478487873e-05\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:34 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:34 INFO 139779977344832] #progress_metric: host=algo-1, completed 6 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 2, \"sum\": 2.0, \"min\": 2}}, \"EndTime\": 1586929834.647439, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 1}, \"StartTime\": 1586929806.3811}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:53 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:53 INFO 139779977344832] #quality_metric: host=algo-1, epoch=2, batch=11 train cross_entropy <loss>=(4.347852097219995)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:53 INFO 139779977344832] #quality_metric: host=algo-1, epoch=2, batch=11 train smooth_l1 <loss>=(0.69386428495797)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:53 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:50:53 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:02 INFO 139779977344832] #quality_metric: host=algo-1, epoch=2, validation mAP <score>=(3.614857000155322e-05)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:02 INFO 139779977344832] Updating the best model with validation-mAP=3.614857000155322e-05\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:03 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:03 INFO 139779977344832] #progress_metric: host=algo-1, completed 10 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 3, \"sum\": 3.0, \"min\": 3}}, \"EndTime\": 1586929863.140469, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 2}, \"StartTime\": 1586929834.647636}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:22 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:22 INFO 139779977344832] #quality_metric: host=algo-1, epoch=3, batch=11 train cross_entropy <loss>=(4.2948799751900335)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:22 INFO 139779977344832] #quality_metric: host=algo-1, epoch=3, batch=11 train smooth_l1 <loss>=(0.6986774930622587)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:22 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:22 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:31 INFO 139779977344832] #quality_metric: host=algo-1, epoch=3, validation mAP <score>=(5.803830770092932e-05)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:31 INFO 139779977344832] Updating the best model with validation-mAP=5.803830770092932e-05\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:31 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:31 INFO 139779977344832] #progress_metric: host=algo-1, completed 13 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 4, \"sum\": 4.0, \"min\": 4}}, \"EndTime\": 1586929891.462843, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 3}, \"StartTime\": 1586929863.140667}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:50 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:50 INFO 139779977344832] #quality_metric: host=algo-1, epoch=4, batch=11 train cross_entropy <loss>=(4.241041150534733)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:50 INFO 139779977344832] #quality_metric: host=algo-1, epoch=4, batch=11 train smooth_l1 <loss>=(0.6813872799955695)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:50 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:50 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:59 INFO 139779977344832] #quality_metric: host=algo-1, epoch=4, validation mAP <score>=(0.0002549927171867939)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:51:59 INFO 139779977344832] Updating the best model with validation-mAP=0.0002549927171867939\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:00 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:00 INFO 139779977344832] #progress_metric: host=algo-1, completed 16 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 5, \"sum\": 5.0, \"min\": 5}}, \"EndTime\": 1586929920.028326, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 4}, \"StartTime\": 1586929891.466318}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:19 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:19 INFO 139779977344832] #quality_metric: host=algo-1, epoch=5, batch=11 train cross_entropy <loss>=(4.179943654444311)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:19 INFO 139779977344832] #quality_metric: host=algo-1, epoch=5, batch=11 train smooth_l1 <loss>=(0.6987006199824346)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:19 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:19 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:28 INFO 139779977344832] #quality_metric: host=algo-1, epoch=5, validation mAP <score>=(0.0007918991246738668)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:28 INFO 139779977344832] Updating the best model with validation-mAP=0.0007918991246738668\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:28 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:28 INFO 139779977344832] #progress_metric: host=algo-1, completed 20 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 6, \"sum\": 6.0, \"min\": 6}}, \"EndTime\": 1586929948.539552, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 5}, \"StartTime\": 1586929920.028523}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:47 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:47 INFO 139779977344832] #quality_metric: host=algo-1, epoch=6, batch=11 train cross_entropy <loss>=(4.107142063244772)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:47 INFO 139779977344832] #quality_metric: host=algo-1, epoch=6, batch=11 train smooth_l1 <loss>=(0.6939098847979238)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:47 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:47 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:56 INFO 139779977344832] #quality_metric: host=algo-1, epoch=6, validation mAP <score>=(0.00146840765284956)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:56 INFO 139779977344832] Updating the best model with validation-mAP=0.00146840765284956\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:56 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:52:56 INFO 139779977344832] #progress_metric: host=algo-1, completed 23 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 7, \"sum\": 7.0, \"min\": 7}}, \"EndTime\": 1586929976.996396, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 6}, \"StartTime\": 1586929948.541594}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:14 INFO 139779977344832] #quality_metric: host=algo-1, epoch=7, batch=10 train cross_entropy <loss>=(4.044046085996178)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:14 INFO 139779977344832] #quality_metric: host=algo-1, epoch=7, batch=10 train smooth_l1 <loss>=(0.6798707888361722)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:14 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:14 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:23 INFO 139779977344832] #quality_metric: host=algo-1, epoch=7, validation mAP <score>=(0.002605024533058733)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:23 INFO 139779977344832] Updating the best model with validation-mAP=0.002605024533058733\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:23 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:23 INFO 139779977344832] #progress_metric: host=algo-1, completed 26 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 8, \"sum\": 8.0, \"min\": 8}}, \"EndTime\": 1586930003.555848, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 7}, \"StartTime\": 1586929976.997282}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:43 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:43 INFO 139779977344832] #quality_metric: host=algo-1, epoch=8, batch=11 train cross_entropy <loss>=(3.94777729613981)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:43 INFO 139779977344832] #quality_metric: host=algo-1, epoch=8, batch=11 train smooth_l1 <loss>=(0.6831795309676605)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:43 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:43 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:52 INFO 139779977344832] #quality_metric: host=algo-1, epoch=8, validation mAP <score>=(0.006021297792739851)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:52 INFO 139779977344832] Updating the best model with validation-mAP=0.006021297792739851\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:52 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:53:52 INFO 139779977344832] #progress_metric: host=algo-1, completed 30 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 9, \"sum\": 9.0, \"min\": 9}}, \"EndTime\": 1586930032.801648, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 8}, \"StartTime\": 1586930003.556052}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:12 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:12 INFO 139779977344832] #quality_metric: host=algo-1, epoch=9, batch=11 train cross_entropy <loss>=(3.83269510089815)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:12 INFO 139779977344832] #quality_metric: host=algo-1, epoch=9, batch=11 train smooth_l1 <loss>=(0.6665820030530554)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:12 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:12 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:21 INFO 139779977344832] #quality_metric: host=algo-1, epoch=9, validation mAP <score>=(0.007843939854808673)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:21 INFO 139779977344832] Updating the best model with validation-mAP=0.007843939854808673\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:21 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:21 INFO 139779977344832] #progress_metric: host=algo-1, completed 33 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 10, \"sum\": 10.0, \"min\": 10}}, \"EndTime\": 1586930061.455676, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 9}, \"StartTime\": 1586930032.802423}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:40 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:40 INFO 139779977344832] #quality_metric: host=algo-1, epoch=10, batch=11 train cross_entropy <loss>=(3.578482235179228)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:40 INFO 139779977344832] #quality_metric: host=algo-1, epoch=10, batch=11 train smooth_l1 <loss>=(0.6668959540973148)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:40 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:40 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:49 INFO 139779977344832] #quality_metric: host=algo-1, epoch=10, validation mAP <score>=(0.011534606096589095)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:49 INFO 139779977344832] Updating the best model with validation-mAP=0.011534606096589095\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:50 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:54:50 INFO 139779977344832] #progress_metric: host=algo-1, completed 36 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 11, \"sum\": 11.0, \"min\": 11}}, \"EndTime\": 1586930090.132682, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 10}, \"StartTime\": 1586930061.455876}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:09 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:09 INFO 139779977344832] #quality_metric: host=algo-1, epoch=11, batch=11 train cross_entropy <loss>=(3.211396718063724)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:09 INFO 139779977344832] #quality_metric: host=algo-1, epoch=11, batch=11 train smooth_l1 <loss>=(0.6526671056793657)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:09 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:09 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:18 INFO 139779977344832] #quality_metric: host=algo-1, epoch=11, validation mAP <score>=(0.010836715532085336)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:18 INFO 139779977344832] #progress_metric: host=algo-1, completed 40 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 12, \"sum\": 12.0, \"min\": 12}}, \"EndTime\": 1586930118.40752, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 11}, \"StartTime\": 1586930090.133552}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:37 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:37 INFO 139779977344832] #quality_metric: host=algo-1, epoch=12, batch=11 train cross_entropy <loss>=(2.8082809749799282)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:37 INFO 139779977344832] #quality_metric: host=algo-1, epoch=12, batch=11 train smooth_l1 <loss>=(0.6596144001474493)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:37 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:37 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:46 INFO 139779977344832] #quality_metric: host=algo-1, epoch=12, validation mAP <score>=(0.01016460956889273)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:55:46 INFO 139779977344832] #progress_metric: host=algo-1, completed 43 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 13, \"sum\": 13.0, \"min\": 13}}, \"EndTime\": 1586930146.616647, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 12}, \"StartTime\": 1586930118.407734}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:05 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:05 INFO 139779977344832] #quality_metric: host=algo-1, epoch=13, batch=11 train cross_entropy <loss>=(2.4184538168985337)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:05 INFO 139779977344832] #quality_metric: host=algo-1, epoch=13, batch=11 train smooth_l1 <loss>=(0.6644528154466973)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:05 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:05 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:14 INFO 139779977344832] #quality_metric: host=algo-1, epoch=13, validation mAP <score>=(0.014552342376822949)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:14 INFO 139779977344832] Updating the best model with validation-mAP=0.014552342376822949\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:14 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:15 INFO 139779977344832] #progress_metric: host=algo-1, completed 46 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 14, \"sum\": 14.0, \"min\": 14}}, \"EndTime\": 1586930175.010897, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 13}, \"StartTime\": 1586930146.61687}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:34 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:34 INFO 139779977344832] #quality_metric: host=algo-1, epoch=14, batch=11 train cross_entropy <loss>=(2.0189563107958026)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:34 INFO 139779977344832] #quality_metric: host=algo-1, epoch=14, batch=11 train smooth_l1 <loss>=(0.6812100937787224)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:34 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:34 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:42 INFO 139779977344832] #quality_metric: host=algo-1, epoch=14, validation mAP <score>=(0.039393026794164426)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:42 INFO 139779977344832] Updating the best model with validation-mAP=0.039393026794164426\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:42 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:42 INFO 139779977344832] #progress_metric: host=algo-1, completed 50 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 15, \"sum\": 15.0, \"min\": 15}}, \"EndTime\": 1586930202.501408, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 14}, \"StartTime\": 1586930175.011138}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:59 INFO 139779977344832] #quality_metric: host=algo-1, epoch=15, batch=10 train cross_entropy <loss>=(1.760574682434993)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:59 INFO 139779977344832] #quality_metric: host=algo-1, epoch=15, batch=10 train smooth_l1 <loss>=(0.6372213933005262)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:59 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:56:59 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:08 INFO 139779977344832] #quality_metric: host=algo-1, epoch=15, validation mAP <score>=(0.024063507185358247)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:08 INFO 139779977344832] #progress_metric: host=algo-1, completed 53 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 16, \"sum\": 16.0, \"min\": 16}}, \"EndTime\": 1586930228.911699, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 15}, \"StartTime\": 1586930202.501588}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:28 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:28 INFO 139779977344832] #quality_metric: host=algo-1, epoch=16, batch=11 train cross_entropy <loss>=(1.6184050028303028)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:28 INFO 139779977344832] #quality_metric: host=algo-1, epoch=16, batch=11 train smooth_l1 <loss>=(0.6645128233242878)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:28 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:29 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:37 INFO 139779977344832] #quality_metric: host=algo-1, epoch=16, validation mAP <score>=(0.023274766383956345)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:37 INFO 139779977344832] #progress_metric: host=algo-1, completed 56 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 17, \"sum\": 17.0, \"min\": 17}}, \"EndTime\": 1586930257.86004, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 16}, \"StartTime\": 1586930228.911981}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:56 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:56 INFO 139779977344832] #quality_metric: host=algo-1, epoch=17, batch=11 train cross_entropy <loss>=(1.4884185540666337)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:56 INFO 139779977344832] #quality_metric: host=algo-1, epoch=17, batch=11 train smooth_l1 <loss>=(0.674403690529026)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:56 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:57:57 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:05 INFO 139779977344832] #quality_metric: host=algo-1, epoch=17, validation mAP <score>=(0.04115612941377907)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:05 INFO 139779977344832] Updating the best model with validation-mAP=0.04115612941377907\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:05 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:05 INFO 139779977344832] #progress_metric: host=algo-1, completed 60 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 18, \"sum\": 18.0, \"min\": 18}}, \"EndTime\": 1586930285.973271, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 17}, \"StartTime\": 1586930257.860249}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:25 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:25 INFO 139779977344832] #quality_metric: host=algo-1, epoch=18, batch=11 train cross_entropy <loss>=(1.3945237601675637)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:25 INFO 139779977344832] #quality_metric: host=algo-1, epoch=18, batch=11 train smooth_l1 <loss>=(0.6473397244975126)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:25 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:25 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:33 INFO 139779977344832] #quality_metric: host=algo-1, epoch=18, validation mAP <score>=(0.09289647487887398)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:33 INFO 139779977344832] Updating the best model with validation-mAP=0.09289647487887398\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:34 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:34 INFO 139779977344832] #progress_metric: host=algo-1, completed 63 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 19, \"sum\": 19.0, \"min\": 19}}, \"EndTime\": 1586930314.167851, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 18}, \"StartTime\": 1586930285.974089}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:53 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:53 INFO 139779977344832] #quality_metric: host=algo-1, epoch=19, batch=11 train cross_entropy <loss>=(1.245890719981133)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:53 INFO 139779977344832] #quality_metric: host=algo-1, epoch=19, batch=11 train smooth_l1 <loss>=(0.6286240680308282)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:53 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:58:53 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:02 INFO 139779977344832] #quality_metric: host=algo-1, epoch=19, validation mAP <score>=(0.07309869842613327)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:02 INFO 139779977344832] #progress_metric: host=algo-1, completed 66 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 20, \"sum\": 20.0, \"min\": 20}}, \"EndTime\": 1586930342.357695, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 19}, \"StartTime\": 1586930314.168051}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:21 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:21 INFO 139779977344832] #quality_metric: host=algo-1, epoch=20, batch=11 train cross_entropy <loss>=(1.1246542153238248)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:21 INFO 139779977344832] #quality_metric: host=algo-1, epoch=20, batch=11 train smooth_l1 <loss>=(0.6542205682321757)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:21 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:21 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:30 INFO 139779977344832] #quality_metric: host=algo-1, epoch=20, validation mAP <score>=(0.08173430913655644)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:30 INFO 139779977344832] #progress_metric: host=algo-1, completed 70 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 21, \"sum\": 21.0, \"min\": 21}}, \"EndTime\": 1586930370.43296, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 20}, \"StartTime\": 1586930342.357982}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:49 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:49 INFO 139779977344832] #quality_metric: host=algo-1, epoch=21, batch=11 train cross_entropy <loss>=(1.0015944472881926)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:49 INFO 139779977344832] #quality_metric: host=algo-1, epoch=21, batch=11 train smooth_l1 <loss>=(0.6436393064611098)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:49 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:49 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:58 INFO 139779977344832] #quality_metric: host=algo-1, epoch=21, validation mAP <score>=(0.08111870513967451)\u001b[0m\n",
      "\u001b[34m[04/15/2020 05:59:58 INFO 139779977344832] #progress_metric: host=algo-1, completed 73 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 22, \"sum\": 22.0, \"min\": 22}}, \"EndTime\": 1586930398.506041, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 21}, \"StartTime\": 1586930370.433205}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:17 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:17 INFO 139779977344832] #quality_metric: host=algo-1, epoch=22, batch=11 train cross_entropy <loss>=(0.9727571261131157)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:17 INFO 139779977344832] #quality_metric: host=algo-1, epoch=22, batch=11 train smooth_l1 <loss>=(0.6221535440218651)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:17 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:17 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:26 INFO 139779977344832] #quality_metric: host=algo-1, epoch=22, validation mAP <score>=(0.06611462003648239)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:26 INFO 139779977344832] #progress_metric: host=algo-1, completed 76 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 23, \"sum\": 23.0, \"min\": 23}}, \"EndTime\": 1586930426.636258, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 22}, \"StartTime\": 1586930398.506262}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:43 INFO 139779977344832] #quality_metric: host=algo-1, epoch=23, batch=10 train cross_entropy <loss>=(0.9147484945355828)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:43 INFO 139779977344832] #quality_metric: host=algo-1, epoch=23, batch=10 train smooth_l1 <loss>=(0.6319247626607328)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:43 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:43 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:52 INFO 139779977344832] #quality_metric: host=algo-1, epoch=23, validation mAP <score>=(0.10210527479466232)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:52 INFO 139779977344832] Updating the best model with validation-mAP=0.10210527479466232\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:52 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:00:53 INFO 139779977344832] #progress_metric: host=algo-1, completed 80 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 24, \"sum\": 24.0, \"min\": 24}}, \"EndTime\": 1586930453.042623, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 23}, \"StartTime\": 1586930426.636538}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:13 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:13 INFO 139779977344832] #quality_metric: host=algo-1, epoch=24, batch=11 train cross_entropy <loss>=(0.9223606388743331)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:13 INFO 139779977344832] #quality_metric: host=algo-1, epoch=24, batch=11 train smooth_l1 <loss>=(0.6068625938601611)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:13 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:13 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:22 INFO 139779977344832] #quality_metric: host=algo-1, epoch=24, validation mAP <score>=(0.0852251214659124)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:22 INFO 139779977344832] #progress_metric: host=algo-1, completed 83 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 25, \"sum\": 25.0, \"min\": 25}}, \"EndTime\": 1586930482.159319, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 24}, \"StartTime\": 1586930453.042831}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:41 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:41 INFO 139779977344832] #quality_metric: host=algo-1, epoch=25, batch=11 train cross_entropy <loss>=(0.9070096554092848)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:41 INFO 139779977344832] #quality_metric: host=algo-1, epoch=25, batch=11 train smooth_l1 <loss>=(0.5945487869276045)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:41 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:41 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:50 INFO 139779977344832] #quality_metric: host=algo-1, epoch=25, validation mAP <score>=(0.08911447405298392)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:01:50 INFO 139779977344832] #progress_metric: host=algo-1, completed 86 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 26, \"sum\": 26.0, \"min\": 26}}, \"EndTime\": 1586930510.316502, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 25}, \"StartTime\": 1586930482.159542}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:09 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:09 INFO 139779977344832] #quality_metric: host=algo-1, epoch=26, batch=11 train cross_entropy <loss>=(0.8576317324132398)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:09 INFO 139779977344832] #quality_metric: host=algo-1, epoch=26, batch=11 train smooth_l1 <loss>=(0.5917382010309643)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:09 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:09 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:18 INFO 139779977344832] #quality_metric: host=algo-1, epoch=26, validation mAP <score>=(0.10125509343913343)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:18 INFO 139779977344832] #progress_metric: host=algo-1, completed 90 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 27, \"sum\": 27.0, \"min\": 27}}, \"EndTime\": 1586930538.52957, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 26}, \"StartTime\": 1586930510.31671}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:37 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:37 INFO 139779977344832] #quality_metric: host=algo-1, epoch=27, batch=11 train cross_entropy <loss>=(0.8498447271738705)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:37 INFO 139779977344832] #quality_metric: host=algo-1, epoch=27, batch=11 train smooth_l1 <loss>=(0.5888376060829735)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:37 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:37 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:46 INFO 139779977344832] #quality_metric: host=algo-1, epoch=27, validation mAP <score>=(0.10060232273103517)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:02:46 INFO 139779977344832] #progress_metric: host=algo-1, completed 93 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 28, \"sum\": 28.0, \"min\": 28}}, \"EndTime\": 1586930566.679449, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 27}, \"StartTime\": 1586930538.529753}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:06 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:06 INFO 139779977344832] #quality_metric: host=algo-1, epoch=28, batch=11 train cross_entropy <loss>=(0.8335050143683412)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:06 INFO 139779977344832] #quality_metric: host=algo-1, epoch=28, batch=11 train smooth_l1 <loss>=(0.6309483235594313)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:06 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:06 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:14 INFO 139779977344832] #quality_metric: host=algo-1, epoch=28, validation mAP <score>=(0.13035674923294702)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:14 INFO 139779977344832] Updating the best model with validation-mAP=0.13035674923294702\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:15 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:15 INFO 139779977344832] #progress_metric: host=algo-1, completed 96 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 29, \"sum\": 29.0, \"min\": 29}}, \"EndTime\": 1586930595.168896, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 28}, \"StartTime\": 1586930566.679658}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:34 WARNING 139779977344832] Expected number of batches: 10, did not match the number of batches processed: 11. This may happen when some images or annotations are invalid and cannot be parsed. Please check the dataset and ensure it follows the format in the documentation.\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:34 INFO 139779977344832] #quality_metric: host=algo-1, epoch=29, batch=11 train cross_entropy <loss>=(0.8337933872984575)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:34 INFO 139779977344832] #quality_metric: host=algo-1, epoch=29, batch=11 train smooth_l1 <loss>=(0.5818220800670001)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:34 INFO 139779977344832] Round of batches complete\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:34 INFO 139779977344832] Updated the metrics\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:43 INFO 139779977344832] #quality_metric: host=algo-1, epoch=29, validation mAP <score>=(0.11984784328228287)\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:43 INFO 139779977344832] #progress_metric: host=algo-1, completed 100 % of epochs\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"Max Batches Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Batches Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Number of Records Since Last Reset\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Batches Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Total Records Seen\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Max Records Seen Between Resets\": {\"count\": 1, \"max\": 0, \"sum\": 0.0, \"min\": 0}, \"Reset Count\": {\"count\": 1, \"max\": 30, \"sum\": 30.0, \"min\": 30}}, \"EndTime\": 1586930623.458663, \"Dimensions\": {\"Host\": \"algo-1\", \"Meta\": \"training_data_iter\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\", \"epoch\": 29}, \"StartTime\": 1586930595.1691}\n",
      "\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:43 WARNING 139779977344832] wait_for_all_workers will not sync workers since the kv store is not running distributed\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:43 INFO 139779977344832] Saved checkpoint to \"/opt/ml/model/model_algo_1-0000.params\"\u001b[0m\n",
      "\u001b[34m[04/15/2020 06:03:43 INFO 139779977344832] Test data is not provided.\u001b[0m\n",
      "\u001b[34m#metrics {\"Metrics\": {\"epochs\": {\"count\": 1, \"max\": 30, \"sum\": 30.0, \"min\": 30}, \"totaltime\": {\"count\": 1, \"max\": 883037.8859043121, \"sum\": 883037.8859043121, \"min\": 883037.8859043121}, \"setuptime\": {\"count\": 1, \"max\": 10.460853576660156, \"sum\": 10.460853576660156, \"min\": 10.460853576660156}}, \"EndTime\": 1586930623.945437, \"Dimensions\": {\"Host\": \"algo-1\", \"Operation\": \"training\", \"Algorithm\": \"AWS/Object Detection\"}, \"StartTime\": 1586929740.982221}\n",
      "\u001b[0m\n",
      "\n",
      "2020-04-15 06:04:11 Uploading - Uploading generated training model\n",
      "2020-04-15 06:04:43 Completed - Training job completed\n",
      "Training seconds: 1034\n",
      "Billable seconds: 1034\n"
     ]
    }
   ],
   "source": [
    "od_model.fit(inputs=data_channels, logs=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "conda_amazonei_tensorflow_p36",
   "language": "python",
   "name": "conda_amazonei_tensorflow_p36"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
